src/applications/vaos/appointment-list/redux/actions.js

```diff
@@ -423,49 +424,100 @@ export function fetchPastAppointments(startDate, endDate, selectedIndex) {
 
 export function fetchRequestDetails(id) {
   return async (dispatch, getState) => {
-    const state = getState();
-    const { appointmentDetails, requestMessages, pending } = state.appointments;
-    const request = appointmentDetails[id] || pending?.find(p => p.id === id);
+    try {
+      const state = getState();
+      let request = selectRequestById(state, id);
+      let facilityId = getVAAppointmentLocationId(request);
+      let facility = state.appointments.facilityData?.[facilityId];
+
+      if (!request || (facilityId && !facility)) {
+        dispatch({
+          type: FETCH_REQUEST_DETAILS,
+        });
+      }
 
-    dispatch({
-      type: FETCH_REQUEST_DETAILS,
-    });
+      if (!request) {
+        request = await fetchRequestById(id);
+        facilityId = getVAAppointmentLocationId(request);
+        facility = state.appointments.facilityData?.[facilityId];
+      }
+
+      if (facilityId && !facility) {
+        try {
```

Comment:
Generally, failing to fetch facility details hasn't been something that throws an error for the whole item, so I'm not dispatching the failure action.

There's some work to clean up the details pages to make sure the lack of facility details doesn't hide all the information.